ci/upload_to_owncloud.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2020 Tomer Eliyahu (Intel)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+scriptdir="$(cd "${0%/*}" && pwd)"
+rootdir=$(realpath "$scriptdir/../")
+
+# shellcheck source=../tools/functions.sh
+. "$rootdir/tools/functions.sh"
+
+usage() {
+    echo "usage: $(basename "$0") [-hv] <user> <password> <remote-path> <local-path>"
+    echo "  options:"
+    echo "      -h|--help - display this help."
+    echo "      -v|--verbose - enable verbosity"
+    echo "      -u|--url - webDAV URL (default https://ftp.essensium.com/owncloud/remote.php/dav/files/USERNAME)"
+    echo
+    echo "  positional arguments:"
+    echo "      remote-path - path in the cloud, relative to the user home to upload"
+    echo "      local-path - local file or folder to upload"
+    echo ""
+    echo "  Credentials are taken from $HOME/.netrc"
+    echo "  Make sure it has a line like this:"
+    echo "    machine ftp.essensium.com login <user> password <password>"
+    echo ""
+}
+
+main() {
+    if ! OPTS=$(getopt -o 'hvu:' --long help,verbose,url: -n 'parse-options' -- "$@"); then
+        echo echo "Failed parsing options." >&2; usage; exit 1
+    fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -h | --help)            usage; exit 0;;
+            -v | --verbose)         export VERBOSE=true; QUIET=; shift;;
+            -u | --owncloud-url)    OWNCLOUD_URL="$2"; shift 2;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1;;
+        esac
+    done
+
+    [ -n "$1" ] || { usage; err "Missing remote-path"; exit 1; }
+    remote_path="$1"; shift
+    [ -n "$1" ] || { usage; err "Missing local-path"; exit 1; }
+    local_path="$1"; shift
+
+    info "upload $local_path to $OWNCLOUD_BROWSE_URL/$remote_path/$(basename $local_path) (using user $user)"
+    find "$local_path" -type d -exec \
+        realpath {} --relative-to="$(dirname "$local_path")" \; | \
+            while read -r -s dir; do
+                printf . # show progress
+                dbg "Create directory: $dir" 
+                curl ${QUIET:+ -s -S} -f -n -X MKCOL "$OWNCLOUD_URL/$user/$remote_path/$dir"
```

Comment:
Can you exit the script with an exit code different than 0 if any of these fail? (not necessarily at the first error, but it would be great to know that something went wrong)
It would allow the job to fail in CI when there is an issue with the upload: https://gitlab.com/prpl-foundation/prplMesh/-/jobs/488726605

Reply:
Done